domains/SignIn.js

```diff
@@ -52,22 +101,118 @@ export default function SignUp({ navigation }) {
               placeholder="johndoe@example.com"
               autoFocus
             />
-            <FormInput
-              label="Password"
-              formikProps={formikProps}
-              formikKey="password"
-              placeholder="Password here"
-              secureTextEntry
-            />
+            {!showPassword ? (
+              <FormInput
+                label="Password"
+                formikProps={formikProps}
+                formikKey="password"
+                placeholder="Password here"
+                secureTextEntry
+              />
+            ) : (
+              <FormInput
+                label="Password"
+                formikProps={formikProps}
+                formikKey="password"
+                placeholder="Password here"
+              />
+            )}
+            <View style={styles.container}>
+              <CheckBox
+                onClick={() => setShowPassword(!showPassword)}
+                isChecked={showPassword}
+                rightText="Show Password"
+              />
+            </View>
             {formikProps.isSubmitting ? (
               <ActivityIndicator />
             ) : (
               <Button title="Submit" onPress={formikProps.handleSubmit} />
             )}
             <Button title="Don't have an account, Sign Up!" onPress={handleSignUp} />
+            <Modal
+              animationType="slide"
+              transparent
+              visible={modalVisible}
+            >
+              <View style={styles.centeredView}>
+                <View style={styles.modalView}>
+                  <Text style={styles.modalText}>
+                    Would you like to use your saved login credentials to sign in?
+                  </Text>
+                  <View style={styles.modalButtons}>
+                    <TouchableHighlight
+                      style={{ ...styles.openButton, backgroundColor: '#2196F3' }}
+                      onPress={() => {
+                        setModalVisible(!modalVisible);
+                        formikProps.values.username = user.username;
+                        formikProps.values.password = user.password;
+                        formikProps.handleSubmit();
+                      }}
+                    >
+                      <Text style={styles.textStyle}>Yes</Text>
+                    </TouchableHighlight>
+                    <TouchableHighlight
+                      style={{ ...styles.openButton, backgroundColor: '#2196F3' }}
+                      onPress={() => {
+                        setModalVisible(!modalVisible);
+                      }}
+                    >
+                      <Text style={styles.textStyle}>No</Text>
+                    </TouchableHighlight>
+                  </View>
+                </View>
+              </View>
+            </Modal>
           </>
         )}
       </Formik>
     </SafeAreaView>
   );
 }
+
+const styles = StyleSheet.create({
```

Comment:
I'm going to create a layout component that should make our styling lives easier, but this will do for now